src/MyConnection.h

```diff
@@ -6,6 +6,10 @@
 #include <boost/noncopyable.hpp>
 #include <boost/shared_ptr.hpp>
 
+const char* as_api_string(const Rcpp::Nullable<std::string>& cpp_string) {
```

Comment:
There are very few applications where I'd use a macro, this is not one of them :-(

I'll take your changes until c1d4e99 and merge manually. Please feel free to push any changes to this PR.